From 144c0863b1ebfad8e229628bffc3551fca851fcb Mon Sep 17 00:00:00 2001
From: Ainy Kumari <ainy.kumari@oss.qualcomm.com>
Date: Thu, 11 Dec 2025 18:44:21 +0530
Subject: [PATCH 07/29] EPPKE: Add wiphy capability flag for EPPKE
 authentication

Introduce a wiphy capability flag to allow the supplicant
to query EPPKE authentication support in STA mode via
NL80211_CMD_GET_WIPHY netlink interface.

Signed-off-by: Ainy Kumari <ainy.kumari@oss.qualcomm.com>
---
 src/drivers/driver.h              | 3 +++
 src/drivers/driver_nl80211_capa.c | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/src/drivers/driver.h b/src/drivers/driver.h
index 172a89aa3..2ed0282eb 100644
--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -2435,6 +2435,9 @@ struct wpa_driver_capa {
 #define WPA_DRIVER_FLAGS2_NON_TRIGGER_BASED_INITIATOR	0x0000000040000000ULL
 /** Driver supports (Re)Association Request/Response frame encryption */
 #define WPA_DRIVER_FLAGS2_ASSOCIATION_FRAME_ENCRYPTION	0x0000000080000000ULL
+/** Driver supports EPPKE authentication */
+#define WPA_DRIVER_FLAGS2_EPPKE				0x0000000100000000ULL
+
 
 	u64 flags2;
 
diff --git a/src/drivers/driver_nl80211_capa.c b/src/drivers/driver_nl80211_capa.c
index 5200ddf8d..480676632 100644
--- a/src/drivers/driver_nl80211_capa.c
+++ b/src/drivers/driver_nl80211_capa.c
@@ -725,6 +725,11 @@ static void wiphy_info_ext_feature_flags(struct wiphy_info_data *info,
 	if (ext_feature_isset(ext_features, len,
 			      NL80211_EXT_FEATURE_ASSOC_FRAME_ENCRYPTION))
 		capa->flags2 |= WPA_DRIVER_FLAGS2_ASSOCIATION_FRAME_ENCRYPTION;
+
+	if (ext_feature_isset(ext_features, len, NL80211_EXT_FEATURE_EPPKE)) {
+		capa->flags2 |= WPA_DRIVER_FLAGS2_EPPKE;
+		capa->flags2 |= WPA_DRIVER_FLAGS2_ASSOCIATION_FRAME_ENCRYPTION;
+	}
 }
 
 
-- 
2.43.0

